modules/mono/glue/GodotSharp/GodotSharp/Variant.cs

```diff
@@ -0,0 +1,841 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Godot.NativeInterop;
+
+namespace Godot;
+
+#nullable enable
+
+[SuppressMessage("ReSharper", "RedundantNameQualifier")]
+public partial struct Variant : IDisposable
+{
+    internal godot_variant.movable NativeVar;
+    private object? _obj;
+    private Disposer? _disposer;
+
+    private class Disposer : IDisposable
```

Comment:
This class can be sealed but not sure if it makes much of a difference since it's private.